### PR TITLE
Add scaffold for E2E tests

### PR DESCRIPTION
**Pull request by mcharriere**
Signed-off-by: Matias Charriere <matias@giantswarm.io>

This PR:

- Adds scaffold for E2E tests

Towards https://github.com/giantswarm/giantswarm/issues/31354

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: tests/e2e/suites/basic/basic_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		WithIsUpgrade(isUpgrade).
+		WithValuesFile("./values.yaml").
+		AfterClusterReady(func() {
+			// no
+		}).
+		BeforeUpgrade(func() {
+			// E.g. ensure that the initial install has completed and has settled before upgrading
+		}).
+		Tests(func() {
+			// we don't do anything here
```
Comment (mcharriere) [inline tests/e2e/suites/basic/basic_suite_test.go:34]: of course we need to, as the basic test is not doing anything valuable.

**Comment by mcharriere** [comment]
There we go. This is only a scaffold to be able to build tests on top.

As you can see, nothing is actually being tested; BUT nothing breaks :)

```
Ran 0 of 2 Specs in 1669.198 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 2 Skipped
PASS
```

**Review by ubergesundheit — APPROVED** [review]
:tada:

**Pull request merged**